### PR TITLE
Added focus handling

### PR DESCRIPTION
**Pull request by domiin50**

### CODE REVIEW

File: src/tracker/FocusListener.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      { setFocus: _setFocus }
+    ),
+    withState('refProp', 'setRef', null),
+    withHandlers({
+      handleRef: ({ setRef }) => ref => {
+        ref && setRef(ref)
+      }
+    }),
+    lifecycle({
+      componentDidUpdate(prevProps) {
```
Comment (domiin50) [inline src/tracker/FocusListener.js]: Handling focus when prop in the state is updated by using a key.

File: src/tracker/FocusListener.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        if (
+          typeof this.props.focus !== 'undefined' &&
+          this.props.focus !== prevProps.focus &&
+          this.props.focus === focus
+        ) {
+          this.props.refProp.focus()
+          this.props.setFocus(undefined)
```
Comment (domiin50) [inline src/tracker/FocusListener.js]: The setting of focus to undefined for handling next focus by a key.

File: src/tracker/Logger/index.js [hunk 3]
```diff
@@ -94,8 +91,13 @@ export default compose(
   withState('description', 'setDescription', ''),
   withState('inputOverride', 'setInputOverride', undefined),
   withHandlers({
-    handleClick: ({ firstSelectedPoint, switchMode }) => () => {
-      if (!firstSelectedPoint) switchMode(MODES.LOGGING)
+    handleFocus: ({ mode, firstSelectedPoint, switchMode }) => () =>
+      mode === MODES.NORMAL && !firstSelectedPoint && switchMode(MODES.LOGGING),
+    handleKeyDown: ({ mode }) => (e: any) => {
```
Comment (domiin50) [inline src/tracker/Logger/index.js]: In one of the next commits, it will be refactored and wrapped into the handler.

File: src/utils.js [hunk 4]
```diff
@@ -36,3 +36,8 @@ export const filterObject = (obj: Object, objFilter: Object): Object =>
     }
     return accum
   }, {})
+
+export const blurElement = () => {
```
Comment (domiin50) [inline src/utils.js]: Function for blurring an active input or element.

File: src/tracker/FocusListener.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+// @flow
+import { connect } from 'react-redux'
+import { compose, withState, withHandlers, lifecycle } from 'recompose'
+
+import { setFocus as _setFocus } from '../actions/trackerActions'
+
+import type { Focus } from '../consts'
+
+const FocusListener = (focus: Focus) =>
```
Comment (mikx98) [inline src/tracker/FocusListener.js]: HOCs generally start with a lower case letter and it's a good practice to prefix them with `with`. So maybe `withFocusListener`?

File: src/tracker/keyBindingsUtils.js [hunk 6]
```diff
@@ -102,19 +101,23 @@ const getKeyBindings = ({
     },
     [MODES.NORMAL]: {
       plainKey: {
+        escape: {
+          description: 'Clear focus',
+          handler: () => blurElement()
```
Comment (mikx98) [inline src/tracker/keyBindingsUtils.js]:
```suggestion
          handler: blurElement
```

**Pull request merged**